### PR TITLE
Add item::get_food

### PR DESCRIPTION
**Pull request by jbytheway**
#### Summary
SUMMARY: None

#### Purpose of change
There were many places in the codebase which had to fetch either an item or its contents, depending on which was food.  This led to code duplication and distraction from the main intent of the code.

#### Describe the solution
Factor that logic out into a new item member function, to simplify code elsewhere.

#### Testing
Unit tests.

#### Additional context
A side-project during my work on #28180, where I also found myself needing this functionality.

### CODE REVIEW

File: src/item.cpp [hunk 1]
```diff
@@ -5721,6 +5714,28 @@ float item::get_freeze_point() const
     return get_comestible()->freeze_point;
 }
 
+item *item::get_food()
+{
+    if( is_food() ) {
```
Comment (BevapDin) [inline src/item.cpp]: You can avoid some code duplication by forwarding to the other getter:
```C++
return const_cast<item*>( const_cast<const item *>( this )->get_food() );
```

Comment (jbytheway) [reply src/item.cpp]: I could, but I don't think it's worth the `const_cast`.  If the code duplication is deemed sufficiently problematic, I'd instead solve that with a template helper function.

Comment (BevapDin) [reply src/item.cpp]: Well, this PR is to get rid of code duplication, so it ... adds duplicated code? A helper function template is also a solution.

Comment (jbytheway) [reply src/item.cpp]: I was on the fence about it, but sure.  Now done.

**Pull request merged**